chainer/gradient_check.py

```diff
@@ -186,6 +186,8 @@ def check_backward(func, x_data, y_grad, params=(),
             :func:`chainer.testing.assert_allclose`.
         no_grads (list of bool): Flag to skip variable for gradient assertion.
             It should be same length as ``x_data``.
+        dtype (~numpy.dtype): `x_data` and `y_grad` are casted to this dtype
+            when calculating numerical gradients. Only float type is allowed.
```

Comment:
If we restrict `dtype` to `float`, I think we should raise some error in other case. Because it can detect unexpected mistakes.